swapi/src/scripts/main.js

```diff
@@ -0,0 +1,57 @@
+import "../styles/main.scss";
+import NavigationView from "./navView";
+import Model from "./dataModel";
+import MainFieldView from "./mainFieldView";
+import PaginationView from "./paginationView";
+
+
+//model.getPeopleData(chooseNavCategory)
+class Controller {
+    constructor() {
+        this.currentCategory = null;
+        this.currentPage = null;
+        this.isNewCategory = true;
+        this.init();
+    }
+
+    init() {
+        this.navView = new NavigationView(this.chooseNavCategoryHandler.bind(this));
+        this.paginationView = new PaginationView(this.getPaginationCount.bind(this));
```

Comment:
why do you pass action in constructors ? constructor used for properties initialization only

Reply:
These actions are callback for ```eventListener```.
```suggestion
class Controller {
  constructor() {
    this.currentCategory = null;
    this.currentPage = null;
    this.isNewCategory = true;
    this.navView = new NavigationView(this.chooseNavCategoryHandler.bind(this));
    this.paginationView = new PaginationView(
      this.getPaginationCount.bind(this)
    );
    this.model = new Model();
    this.mainFieldView = new MainFieldView();
  }
```